src/multiselect_draggable.js

```diff
@@ -0,0 +1,281 @@
+   * It delegates the startDrag methods to the topmost subdraggables.
+   * @param {Blockly.Events.BLOCK_DRAG} e A drag event
+   */
+  startDrag(e) {
+    this.topSubDraggables = [];
+    for (const draggable of this.subDraggables) {
+      // Save only the topmost subdraggables to do the drag on
+      if (!(hasSelectedParent(draggable[0], true))) {
+        this.topSubDraggables.push(draggable[0]);
+      }
+
+      // Save any connections between blocks of the same level (next statement)
+      const parentBlock = draggable[0].getParent();
+      if (parentBlock && this.subDraggables.has(parentBlock) &&
+          parentBlock.getNextBlock() === draggable[0]) {
+        this.connectionDBList.push([parentBlock.nextConnection,
+          draggable[0].previousConnection]);
+      }
+
+      this.subDraggables.set(draggable[0],
+          draggable[0].getRelativeToSurfaceXY());
+    }
+
+    for (const draggable of this.topSubDraggables) {
+      draggable.startDrag();
+    }
+  }
+
+  /**
+   * Drags the subdraggables of the multiselectDraggable object if
+   * there are no other plugins that overwrite this method.
+   * It delegates the drag methods to the topmost subdraggables.
+   * @param {Blockly.utils.Coordinate} newLoc The new location the object
+   * is being moved to (in Blockly coordinates)
+   * @param {Blockly.Events.BLOCK_DRAG} e A drag event
+   */
+  drag(newLoc, e) {
+    for (const draggable of this.topSubDraggables) {
+      draggable.drag(Blockly.utils.Coordinate.sum(newLoc,
+          this.subDraggables.get(draggable)), e);
+    }
+  }
+
+  /**
+   * Ends the drag of the subdraggables of the multiselectDraggable
+   * object if there are no other plugins that overwrite this method.
+   * It delegates the endDrag methods to the topmost subdraggables.
+   * @param {Blockly.Events.BLOCK_DRAG} e A drag event
+   */
+  endDrag(e) {
+    for (const draggable of this.topSubDraggables) {
+      draggable.endDrag(e);
+    }
+
+    for (const draggable of this.subDraggables) {
+      // Update location of subdraggables
+      draggable[1] = draggable[0].getRelativeToSurfaceXY();
+    }
+    // Reconnect any connections between blocks that are
+    // on the same level (not child/parent)
+    this.connectionDBList.forEach(function(connectionDB) {
+      connectionDB[0].connect(connectionDB[1]);
+    });
+    this.connectionDBList = [];
+  }
+
+  /**
+   * Reverts any drags done on the multiselectDraggable object.
+   */
+  revertDrag() {
+    for (const draggable of this.subDraggables) {
+      draggable[0].revertDrag();
+    }
+  }
+
+  // ISelectable methods
+  /**
+   * An empty function that is required for the
+   * multiselectDraggable object to be selectable.
+   */
+  select() {
+    // This just needs to be an empty function
+  }
+
+  /**
+   * An empty function that is required for the
+   * multiselectDraggable object to be selectable.
+   */
+  unselect() {
+    // This just needs to be an empty function
+  }
```

Comment:
I just noticed that we still use `element.pathObject.updateSelected(false);` & `element.pathObject.updateSelected(true);` here and there, can we just put these things here? Maybe we will get hooked calls automatically (according to my gut feeling), and then possibly we can get rid of these manual updates. (I don't remember exactly but you seem to have told me that something will not work here without those manual updates, so ignore this one if this is not possible).

Anyway, I think the management of getting a block visually selected is Blockly's job as now they introduced MultiDraggable, we shouldn't still do these calls on our side.

Reply:
If we put those statements in the select() method, it only updates the selected highlight after the multiselection mode is turned off.

Reply:
Ah, now I remembered. So did you investigate why it only updates the selected highlight after the multiselection mode is turned off if we do that way? Is that because we pass these updates to Blockly core only after we turn off the multi-selection mode? Have you tried to set MultiselectDraggable already, instead of only adding/removing the ID to the blockSelection set, when we are in multi-selection mode?

Reply:
BTW, any more comments for this one? @changminbark

> Ah, now I remember. So did you investigate why it only updates the selected highlight after the multiselection mode is turned off if we do that way? Is that because we pass these updates to Blockly core only after we turn off the multi-selection mode? Have you tried to set MultiselectDraggable already, instead of only adding/removing the ID to the blockSelection set, when we are in multi-selection mode?

Reply:
I did not dive too deep into why it only updates after the multiselection mode is turned off. However, I think that it may be because of the way the eventListenerAll_ works in the multiselect_controls file. Apparently, the Blockly selected event is fired multiple times even with one action of selecting. I am not sure why this happens but I am already setting the selected as the MultiselectDraggable already in the updateMultiselect function, which is the function called in the eventListenerAll_.

Reply:
Hmm, in any case, it will be good if we manage to use the `select()` and `unselect()` here to get rid of those manual updates of `element.pathObject.updateSelected`. 

> the Blockly selected event is fired multiple times even with one action of selecting. 

Can that be a similar cascading issue just like #58? Feel free to modify the eventListenerAll_ in the multiselect_controls file to get this fixed.

Reply:
I already have the updateMultiselect function take into account these events that get fired.